pkg/handler/enqueue_mapped.go

```diff
@@ -25,6 +25,22 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
 )
 
+// FromMapFunc enqueues Requests by running a transformation function that outputs a collection
+// of reconcile.Requests on each Event.  The reconcile.Requests may be for an arbitrary set of objects
+// defined by some user specified transformation of the source Event.  (e.g. trigger Reconciler for a set of objects
+// in response to a cluster resize event caused by adding or deleting a Node)
+//
+// FromMapFunc is frequently used to fan-out updates from one object to one or more other
+// objects of a differing type.
+//
+// For UpdateEvents which contain both a new and old object, the transformation function is run on both
+// objects and both sets of Requests are enqueue.
+func FromMapFunc(mapFN func(MapObject) []reconcile.Request) EventHandler {
+	return &EnqueueRequestsFromMapFunc{
+		ToRequests: ToRequestsFunc(mapFN),
+	}
+}
+
 var _ EventHandler = &EnqueueRequestsFromMapFunc{}
```

Comment:
Alternatively, we can completely unexport `EnqueueRequestsFromMapFunc`. It doesn't provide any advantage over using `FromMapFunc` and is a lot more annoying to construct

Reply:
+1 to that, better have a single way to use it than two and confuse users

Reply:
Let's make that a ⚠️

Reply:
Updated, PTAL